src/app/routing-module/navigation.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import {ActivatedRoute, Router} from "@angular/router";
+import {ProjectRepositoryService} from "../repository/project-repository.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NavigationService {
+
+  constructor(private router: Router,
+              private route: ActivatedRoute,
+              private projectService: ProjectRepositoryService) {
+
+  }
+
+  /**
+   * navigates to a specific project
+   * @param projectId: The project to which is navigated.
+   */
+  public navigateToProjectsPage() {
+    this.router.navigate(['projects'])
+  }
+
+  /**
+   * navigates to a specific project
+   * @param projectId: The project to which is navigated.
+   */
+  public navigateToProject(projectId: string) {
+    return this.projectService.isProjectExisting(projectId).then( exists => {
+      if (exists) {
+        return this.router.navigate(["project/" + projectId]);
+      } else {
+        console.warn(`Project not found. There is no project with id ${projectId}.`)
+        return;
```

Comment:
If you use Promise you can make use of .catch() or .error() and handle everything there

Reply:
see above